canard.c

```diff
@@ -151,29 +150,19 @@ int16_t canardBroadcast(CanardInstance* ins,
     uint32_t can_id = 0;
     uint16_t crc = 0xFFFFU;
 
+    can_id = ((uint32_t) priority << 26U) | ((uint32_t) data_type_id << 8U) | (uint32_t) canardGetLocalNodeID(ins);
+
     if (canardGetLocalNodeID(ins) == 0)
```

Comment:
Node ID of zero is no longer a special case.

Reply:
What should be done instead? Should only bit 24 be set, remove the inner if (with error) and move the inner if in else out of the if/else statement?

Reply:
We just need to reserve some other node ID value at the implementation level. For example, we could treat every node ID above 127 as "anonymous". The discriminator is now also gone, instead we have "pseudo-ID" now (which is semantically similar to discriminator), see 4.4.1 https://new.uavcan.org/specification/UAVCAN_Specification_v1.0.pdf

Reply:
I must have missed that section earlier... Will add a check which treats every node ID above 127 as anonymous as you proposed along with the pseudo-ID.

Reply:
Not directly the scope of this PR, but during my work of updating the error handling related to anonymous transfers, I came over the error code CANARD_ERROR_NODE_ID_NOT_SET. This is currently only used if the user tries to transmit a multi-frame message or service request/response while in anonymous mode. Previously, this error was descriptive as "missing" node ID value was the cause. However, with the new specification this is no longer the truth as a node can have set an ID, but still be in "anonymous mode".

In my opinion, it should, therefore, be renamed to a more descriptive name e.g. CANARD_ERROR_INVALID_OPERATION_ANONYMOUS_NODE or something similar. Any comments to this?

Reply:
> Previously, this error was descriptive as "missing" node ID value was the cause. However, with the new specification this is no longer the truth as a node can have set an ID, but still be in "anonymous mode".

This semantic was not supposed to be changed. Could you please elaborate how a node can be in the passive (anonymous) mode while having a valid node ID?

I should also note that the default node ID value should be changed: it used to be zero (anonymous), now it should be any value above 127.

Reply:
What I meant is that previously the default node ID was 0 and this meant that it would transmit in anonymous mode unless the nodeID was updated with a non-zero value. With my previous comment, I forgot that we also need to change the default node ID from zero to a value above 127. Hence, a node could transmit in "normal" (non-anonymous) mode even though the node ID wasn't set by the user. However, when the default value is changed to a value above 127 (I propose 255), this is no longer the case, and my previous comment becomes "invalid".

Reply:
Should we treat all node IDs above 127 as anonymous, or restrict it to a single value? After looking at it, I would propose to initialize the node ID to a value above 127 (let's say 255) and then only allow the user to update the value if it is between 0 and 127. This will be similar to how it is done today in canardSetLocalNodeID. In case an invalid value is provided, we can then return an error code.

Reply:
Agree.

Reply:
The above changes have now been implemented, ready for a new review.